exercises/dominoes/.meta/src/reference/java/ChainNotFoundException.java

```diff
@@ -0,0 +1,5 @@
+class ChainNotFoundException extends RuntimeException {
```

Comment:
This should extend `Exception` not `RuntimeException`. When you extend `RuntimeException` it means your custom exception will be [unchecked](https://www.geeksforgeeks.org/checked-vs-unchecked-exceptions-in-java/) so the user doesn't have to deal with it. We want the user to have to deal with the error, therefore our exception should be [checked](https://www.geeksforgeeks.org/checked-vs-unchecked-exceptions-in-java/).

Unchecked exceptions are used for errors that you in most circumstances can't recover from, e.g. out of memory exception. It makes sense to not make the user deal with those kinds of error because usually the best you can do in those cases is just to let the program crash. I hope that makes sense, let me know if not 🙂